src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,87 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.storage.Storage;
+import duke.task.*;
+import duke.ui.Ui;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class AddCommand extends Command {
+    private String type;
+    private String description;
+
+    private static final String MESSAGE_ADD      = "Got it. I've added this task:\n  %s\n"
+            + "Now you have %d %s in the list.";
+
+    private static final String ERROR_MISSING_DESCRIPTION = "The description cannot be empty.";
+    private static final String ERROR_MISSING_DESCRIPTION_AND_TIME = "The description and time cannot be empty.";
+    private static final String ERROR_MISSING_DEADLINE = "The deadline must be present. e.g. task /by Monday";
+    private static final String ERROR_MISSING_EVENT_TIME = "The event time must be present. e.g. meeting /at Monday";
+    private static final String ERROR_WRONG_DATE_FORMAT  = "The date time provided is in the wrong format. "
+            + "Expected d/m/yyyy hh:mm.";
+
+    private final static DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("d/M/uuuu HH:mm");
+
+    public AddCommand(String type, String description) {
+        this.type = type;
+        this.description = description;
+    }
+
+    public void execute(TaskList tasks, Storage storage) throws DukeException {
+        Task task;
+        switch (type) {
+        case "event": {
+            if (!description.matches(".+\\s/at\\s.+$")) {
```

Comment:
I find these lines quite confusing, because of all the symbols. Perhaps you could break it down to make it more readable for people who don't code as well!

Reply:
This is actually regex so I don't think there's a way to make it more readable haha, I'll add a comment to describe what it does, should make it clearer that way.

Regarding the error messages being displayed as final variables, I think it's a matter of choice, I chose to put them as constants as I wanted them to be easily editable, so changing the message can just be done at the top of the file. I actually hope to refactor this into a separate file or class so that it is possible to edit all error and messages in one file, but not sure whether this is even possible.